### PR TITLE
Issue 54: Fixed pydocstyle/flake8-errors in utils.py.

### PR DESCRIPTION
**Pull request by notger**
# Pull Request

## Description

Fixed pydocstyle and flake8 errors for the satip-files.

I also took the liberty of updating the requirements (might have missed something) and flagging potential improvements and outdated, unused public functions where I spotted them (see "TODO"-marked lines in the code.)

Fixes issue #54 

## How Has This Been Tested?

Tests performed by running `pre-commit run --all-files` and `pytest tests`.

- [x] Yes, all tests passed.

Changes do not affect data processing to the best of my knowledge. Quick code check performed, but a more extensive test by running `scripts/generate_test_plots` will be performed once I gained an EUMETSAT-API-key.

I considered a more extensive test run not necessary as nearly all changes were adding/reshaping comments.
The only function code changed was exception handling.

- [x] Yes, sanity checks performed
- [x] No, no extensive test run performed, see above

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~ (no tests for comments)
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by notger** [comment]
@jacobbieker I would now consider this reviewable.

The two tests which fail above do so b/c I took the liberty to amend the requirements file and `cartopy` and `satpy` seem to cause some trouble. (Can confirm this from my local dev machine, where both were causing hick-ups as well.)

I do not consider this within the scope of the ticket, but rather a separate issue to be looked into.

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,13 @@
+bottleneck
+cartopy
+cython
+fsspec
+numcodecs
 numpy>=1.19.4
 pandas>=1.1.4
+rasterio
```
Comment (jacobbieker) [inline requirements.txt:8]: I would go with not including rasterio and cartopy in the requirements, and I don't think cython is needed. I know they are used in the generate_test_plots script, but that is only for github actions really, and as you found out, cartopy, rasterio, and satpy don't all work well together and are a bit hard to get working in a single environment.

File: satip/eumetsat.py [hunk 2]
```diff
@@ -181,15 +187,28 @@ def identify_available_datasets(
...
-        f"https://api.eumetsat.int/data/download/collections/{urllib.parse.quote(collection_id)}/products/{urllib.parse.quote(data_id)}"
+        "https://api.eumetsat.int/data/download/collections/"
+        + f"{urllib.parse.quote(collection_id)}/products/{urllib.parse.quote(data_id)}"
         + "?access_token="
         + access_token
     )
 
 
-def json_extract(json_obj: Union[dict, list], locators: list):
+# TODO: This function is not used anywhere; can be removed.
```
Comment (jacobbieker) [inline satip/eumetsat.py:210]: I would say feel free to remove this one!

File: satip/eumetsat.py [hunk 3]
```diff
@@ -198,19 +217,22 @@ def json_extract(json_obj: Union[dict, list], locators: list):
...
 
 
+# TODO: Remove this class if `check_valid_requests` is removed as well.
+class InvalidCredentials(Exception):
+    """Class to catch the case of invalid credentials."""
+
+    pass
+
+
+# TODO: Remove this function, as its only functionality can also be achieved with raise_for_status
```
Comment (jacobbieker) [inline satip/eumetsat.py:227]: Same here, feel free to remove it! Less code is always better, and there has been a few things left from a large refactoring a few months ago.

File: satip/eumetsat.py [hunk 4]
```diff
@@ -276,42 +297,38 @@ def __init__(
...
+        Args:
             data_link: Url link for the relevant dataset
-
         """
 
         params = {"access_token": self.access_token}
 
         r = requests.get(data_link, params=params)
-        check_valid_request(r)
+        check_valid_request(r)  # TODO: This could be replaced with the canonical raise_for_status
```
Comment (jacobbieker) [inline satip/eumetsat.py:331]: Sounds good!

File: satip/eumetsat.py [hunk 5]
```diff
@@ -540,7 +556,8 @@ def _download_single_tailored_dataset(
                 )
 
 
-def get_dir_size(directory="."):
+# TODO: Delete? This function is not used anywhere.
```
Comment (jacobbieker) [inline satip/eumetsat.py:559]: Sounds good!

**Review by jacobbieker — COMMENTED** [review]
This looks great! Thanks for going through all of it! There are just a few comments, mostly being feel free to delete unused functions. And yeah, for the broken tests, removing cartopy and rasterio should let them pass, and keeps the dependencies of install satip quite a bit smaller.

**Comment by notger** [comment]
Thanks for the comments! Removed the functions and pruned requirements as suggested.
(Will not mark your comments as "resolved", as that decision is up to you. Can do, if you want.)

**Review by jacobbieker — APPROVED** [review]
Great! Looks good! Merge it when you want! (If you can, I think you should be able to)

**Pull request merged**